src/morphology_workflows/repair.py

```diff
@@ -412,8 +439,13 @@ def make_release(df, _, zero_diameter_path, unravel_path, repair_path, extension
 
         _m = []
         with Pool() as pool:
-            for name, row, m in pool.imap(__create_db_row, df.loc[df["is_valid"]].iterrows()):
-                df.loc[name] = pd.Series(row)
+            for index, row, m in tqdm(
+                pool.imap(__create_db_row, df_tmp.loc[df_tmp["is_valid"]].iterrows()),
+                total=len(df_tmp),
+            ):
+                if index in df.index:
+                    df.loc[index] = pd.Series(row)
+                df_tmp.loc[index] = pd.Series(row)
                 _m.append(m)
```

Comment:
I think you can avoid reseting the index of `df` with something like:
```python
        _m = []
        written = set()
        with Pool() as pool:
            for index, row, m in tqdm(
                pool.imap(__create_db_row, df_tmp.loc[df_tmp["is_valid"]].iterrows()),
                total=len(df_tmp),
            ):
                if row["morph_name"] in df.index and row["morph_name"] not in written:
                    df.loc[row["morph_name"]] = pd.Series(row)
                    written.add(row["morph_name"])
                df_tmp.loc[index] = pd.Series(row)
                _m.append(m)
```

Reply:
baaaa

Reply:
Ahah yeah I agree :laughing: